src/raw_field.rs

```diff
@@ -126,3 +159,24 @@ macro_rules! raw_field_tuple {
         }
     }};
 }
+
+/// Computes a const raw pointer to the given field of the given base pointer
+/// to the given parent tuple typle.
+///
+/// The `base` pointer *must not* be dangling, but it *may* point to
+/// uninitialized memory.
+#[macro_export(local_inner_macros)]
+macro_rules! raw_field_union {
```

Comment:
This is exactly like `raw_field`, except for the field check, right?

Reply:
Yes, exactly.
I can try to reuse the same `raw_field` macro and parameterize by having the caller specify `@union` (give or take some bike shedding);
Is it possible to pass a macro name to another macro?

Reply:
Yes it is, but I don't think I like higher-order macros.^^

`raw_field` is public API I think. So copy-pasting is fine but please add a comment saying that this is exactly the same except for the field check.